cmake/CCCLGetDependencies.cmake

```
@@ -22,6 +22,11 @@ macro(cccl_get_fmt)
   CPMAddPackage("gh:fmtlib/fmt#11.0.1")
 endmacro()
 
+macro(cccl_get_json)
+  include("${_cccl_cpm_file}")
+  CPMAddPackage("gh:nlohmann/json@3.11.3")
```

```suggestion
  CPMAddPackage("gh:nlohmann/json@3.12.0")
```

We can upgrade to the latest released version. 
I verified that the reduce-specific tests from the CUB test suite run fine locally with 3.12.0,